src/EnergyPlus/PhotovoltaicThermalCollectors.cc

```diff
@@ -165,60 +168,96 @@ namespace PhotovoltaicThermalCollectors {
+        bool ErrorsFound(false); // Set to true if errors in input, fatal at end of routine
+        using ScheduleManager::GetScheduleIndex;
+
+        tmpSimplePVTperf.allocate(NumSimplePVTPerform);
+        for (Item = 1; Item <= NumSimplePVTPerform; ++Item) {
+            state.dataInputProcessing->inputProcessor->getObjectItem(state,
+                                                                     state.dataIPShortCut->cCurrentModuleObject,
+                                                                     Item,
+                                                                     state.dataIPShortCut->cAlphaArgs,
+                                                                     NumAlphas,
+                                                                     state.dataIPShortCut->rNumericArgs,
+                                                                     NumNumbers,
+                                                                     IOStatus,
+                                                                     _,
+                                                                     state.dataIPShortCut->lAlphaFieldBlanks,
+                                                                     state.dataIPShortCut->cAlphaFieldNames,
+                                                                     state.dataIPShortCut->cNumericFieldNames);
+            if (UtilityRoutines::IsNameEmpty(state, state.dataIPShortCut->cAlphaArgs(1), state.dataIPShortCut->cCurrentModuleObject, ErrorsFound))
+                continue;
+            tmpSimplePVTperf(Item).Name = state.dataIPShortCut->cAlphaArgs(1);
+            if (UtilityRoutines::SameString(state.dataIPShortCut->cAlphaArgs(2), "Fixed")) {
+                tmpSimplePVTperf(Item).ThermEfficMode = ThermEfficEnum::FIXED;
+            } else if (UtilityRoutines::SameString(state.dataIPShortCut->cAlphaArgs(2), "Scheduled")) {
+                tmpSimplePVTperf(Item).ThermEfficMode = ThermEfficEnum::SCHEDULED;
+            } else {
+                ShowSevereError(state, "Invalid " + state.dataIPShortCut->cAlphaFieldNames(2) + " = " + state.dataIPShortCut->cAlphaArgs(2));
+                ShowContinueError(state, "Entered in " + state.dataIPShortCut->cCurrentModuleObject + " = " + state.dataIPShortCut->cAlphaArgs(1));
+                ErrorsFound = true;
+            }
+            tmpSimplePVTperf(Item).ThermalActiveFract = state.dataIPShortCut->rNumericArgs(1);
+            tmpSimplePVTperf(Item).ThermEffic = state.dataIPShortCut->rNumericArgs(2);
+            tmpSimplePVTperf(Item).ThermEffSchedNum = ScheduleManager::GetScheduleIndex(state, state.dataIPShortCut->cAlphaArgs(3));
+            if ((tmpSimplePVTperf(Item).ThermEffSchedNum == 0) && (tmpSimplePVTperf(Item).ThermEfficMode == ThermEfficEnum::SCHEDULED)) {
+                ShowSevereError(state, "Invalid " + state.dataIPShortCut->cAlphaFieldNames(3) + " = " + state.dataIPShortCut->cAlphaArgs(3));
```

Comment:
Use `format` instead of string concatenation. Many examples in the code.

Reply:
updated to use format